lib/msf/core/module/failure.rb

```diff
@@ -56,9 +56,14 @@ def report_failure
     info[:target_name] = self.target.name if self.respond_to?(:target)
 
     if self.datastore['RHOST'] && (self.options['RHOST'] || self.options['RHOSTS'])
-      info[:host] = self.datastore['RHOST']
+      # Only include RHOST if it's a single valid host, not a multi-value string or file path
+      rhost = self.datastore['RHOST'].to_s
+      # Check if RHOST is not a file path and doesn't contain spaces (multiple hosts)
+      unless rhost.start_with?('file:') || rhost.include?(' ')
```

Comment:
I think if we want to validate that the `rhost` string represents an IP address which is what ActiveRecord needs for the `host` field, we should use `Rex::Socket.is_ip_addr?` here. If `host` has to be an IPv4 address though, we'll need to use the more specific `Rex::Socket.is_ipv4?`.

```suggestion
      if Rex::Socket.is_ip_addr?(rhost)
```